packages/orbit-components/src/Stack/index.tsx

```diff
@@ -1,100 +1,87 @@
-    inline = false,
-    spacing = SPACINGS.MEDIUM,
-    align = ALIGNS.START,
-    legacy = false,
-    justify = JUSTIFY.START,
-    grow = true,
-    wrap = false,
-    shrink = false,
-    basis,
-    spaceAfter,
-    children,
-    mediumMobile,
-    largeMobile,
-    tablet,
-    desktop,
-    largeDesktop,
-    as = "div",
-  } = props;
+  const { children, as: ComponentTag = "div", dataTest, ...restProps } = props;
+
+  const defaultMediaProps = React.useMemo(() => {
+    const {
+      spacing = SPACING.medium,
+      grow = true,
+      inline = false,
+      justify = JUSTIFY.START,
+      shrink = false,
+      wrap = false,
+      direction = DIRECTION.ROW,
+      align = ALIGN.START,
+      spaceAfter,
+    } = restProps;
+
+    return { spacing, grow, inline, justify, align, shrink, wrap, direction, spaceAfter };
+  }, [restProps]);
 
-  const isFlex = shouldUseFlex(props);
-  const direction = props.direction || (isFlex ? DIRECTIONS.ROW : DIRECTIONS.COLUMN);
+  const getTailwindTokensForMedia = React.useCallback(
+    (properties?: CommonProps & Common.SpaceAfter, viewport?: Devices): string => {
+      if (!properties) return "";
 
-  const smallMobile = {
-    direction,
-    align,
-    justify,
-    wrap,
-    grow,
-    basis,
-    inline,
-    shrink,
-    spacing,
-    spaceAfter,
-  };
+      const { spaceAfter, direction, inline, wrap, grow, shrink, align, justify, spacing } =
+        properties;
+
+      return cx(
+        typeof spaceAfter !== "undefined" && getSpaceAfterClasses(spaceAfter, viewport),
+        typeof inline !== "undefined" && getInlineClass(inline, viewport),
+        typeof wrap !== "undefined" && getWrapClasses(wrap, viewport),
+        typeof grow !== "undefined" && getGrowClasses(grow, viewport),
+        typeof shrink !== "undefined" && getShrinkClasses(shrink, viewport),
+        typeof align !== "undefined" && getAlignItemsClasses(align, viewport),
+        typeof align !== "undefined" && getAlignContentClasses(align, viewport),
+        typeof justify !== "undefined" && getJustifyClasses(justify, viewport),
+        typeof spacing !== "undefined" && getSpacingClasses(spacing, viewport, direction),
+        typeof direction !== "undefined" && getDirectionClasses(direction, viewport),
+        inline === false && "w-full",
+      );
+    },
+    [],
+  );
 
   return (
-    <StyledStack
-      dataTest={dataTest}
-      legacy={legacy}
-      flex={isFlex}
-      smallMobile={smallMobile}
-      mediumMobile={mediumMobile}
-      largeMobile={largeMobile}
-      tablet={tablet}
-      desktop={desktop}
-      largeDesktop={largeDesktop}
-      element={as}
+    // @ts-expect-error orbit string as
+    <ComponentTag
+      data-test={dataTest}
+      className={cx(
+        getTailwindTokensForMedia(defaultMediaProps),
+        viewports.map(viewport => {
+          if (!props[viewport]) return null;
+          return getTailwindTokensForMedia(props[viewport], viewport);
+        }),
```

Comment:
It can be memoized, but I am not a big fan of putting memos everywhere. As it makes sense to put only in case of the really slow component which has complicated calculations inside, which is not our case here. It will have zero effect or maybe even worse (it comes with a cost of memory usage). I am considering even removing it above (as it was copied from Search implementation)

Reply:
Agreed. I applied it in Box but I didn't perform any test to see if it is worth to have it. I think we either apply on both or remove or both, after your findings on it 🙂